lib/modules/manager/pip_requirements/extract.ts

```diff
@@ -24,38 +24,44 @@ const specifierPartPattern = `\\s*${rangePattern.replace(
 const specifierPattern = `${specifierPartPattern}(?:\\s*,${specifierPartPattern})*`;
 export const dependencyPattern = `(${packagePattern})(${extrasPattern})(${specifierPattern})`;
 
-export function extractPackageFile(
-  content: string,
-  _: string,
-  config: ExtractConfig
-): PackageFile | null {
+export function cleanRegistryUrls(registryUrls: string[]): string[] {
+  return registryUrls.map((url) => {
+    // handle the optional quotes in eg. `--extra-index-url "https://foo.bar"`
+    const cleaned = url.replace(regEx(/^"/), '').replace(regEx(/"$/), '');
+    if (!GlobalConfig.get('exposeAllEnv')) {
+      return cleaned;
+    }
+    // interpolate any environment variables
+    return cleaned.replace(
+      regEx(/(\$[A-Za-z\d_]+)|(\${[A-Za-z\d_]+})/g),
+      (match) => {
+        const envvar = match
+          .substring(1)
+          .replace(regEx(/^{/), '')
+          .replace(regEx(/}$/), '');
+        const sub = process.env[envvar];
+        return sub || match;
+      }
+    );
+  });
+}
+
+export function extractPackageFile(content: string): PackageFile | null {
   logger.trace('pip_requirements.extractPackageFile()');
 
-  let indexUrl: string | undefined;
-  const extraUrls: string[] = [];
+  let registryUrls: string[] = [];
+  const additionalRegistryUrls: string[] = [];
   content.split(newlineRegex).forEach((line) => {
     if (line.startsWith('--index-url ')) {
-      indexUrl = line.substring('--index-url '.length).split(' ')[0];
+      registryUrls = [line.substring('--index-url '.length).split(' ')[0]];
     }
     if (line.startsWith('--extra-index-url ')) {
       const extraUrl = line
         .substring('--extra-index-url '.length)
         .split(' ')[0];
-      extraUrls.push(extraUrl);
+      additionalRegistryUrls.push(extraUrl);
     }
   });
-  let registryUrls: string[] = [];
-  if (indexUrl) {
-    // index url in file takes precedence
-    registryUrls.push(indexUrl);
-  } else if (config.registryUrls?.length) {
-    // configured registryURls takes next precedence
-    registryUrls = registryUrls.concat(config.registryUrls);
-  } else if (extraUrls.length) {
-    // Use default registry first if extra URLs are present and index URL is not
-    registryUrls.push(process.env.PIP_INDEX_URL || 'https://pypi.org/pypi/');
```

Comment:
don't we loose the pip index url from ENV now?

Reply:
I think it should still work thanks to https://github.com/renovatebot/renovate/blob/c4bd2b27f09a2c64eab62de76b64bcc688c6004d/lib/modules/datasource/pypi/index.ts#L25-L27